tensorrt_llm/llmapi/llm_args.py

```
@@ -3424,6 +3426,92 @@ def get_executor_config(
+    prefill/decode thresholds from the checkpoint config.json and build SkipSoftmaxAttentionConfig.
+    """
+    if data.get("threshold_scale_factor") is not None:
+        return BaseSparseAttentionConfig.from_dict(data)
+
+    algorithm = (data.get("algorithm") or "").lower()
+    target_sparsity_ratio = data.get("target_sparsity_ratio")
+
+    if algorithm == "skip_softmax" and target_sparsity_ratio is not None:
+        if not model_path or not str(model_path).strip():
+            raise ValueError(
+                "Model path is required when using target_sparsity_ratio for skip softmax attention.")
```

Ruff (0.15.4)

[warning] Avoid specifying long messages outside the exception class

(TRY003)